java/client/src/main/java/glide/api/commands/GenericBaseCommands.java

```diff
@@ -269,4 +271,52 @@ CompletableFuture<Boolean> pexpireAt(
      * }</pre>
      */
     CompletableFuture<Long> ttl(String key);
+
+    /**
+     * Invokes a Lua script with its keys and arguments. <br>
+     * This method simplifies the process of invoking scripts on a Redis server by using an object
+     * that represents a Lua script. The script loading, argument preparation, and execution will all
+     * be handled internally. If the script has not already been loaded, it will be loaded
+     * automatically using the Redis <code>SCRIPT LOAD</code> command. After that, it will be invoked
+     * using the Redis <code>EVALSHA</code> command.
+     *
+     * @see <a href="https://redis.io/commands/script-load/">SCRIPT LOAD</a> and <a
+     *     href="https://redis.io/commands/evalsha/">EVALSHA</a> for details.
+     * @param script The Lua script to execute.
+     * @return a value that depends on the script that was executed.
+     * @example
+     *     <pre>{@code
+     * Script luaScript = new Script("return 'Hello'");
+     * String result = (String) client.invokeScript(luaScript).get();
+     * assert result == "Hello";
+     * luaScript.close(); // drops the lua script
```

Comment:
Is it not automatically closed?

Reply:
Not automatically.   It will get closed with garbage collection, but the proper handling of autocloseable is either directly using the .close() call, or by wrapping it in a `try() {}` block.